### DISCUSSION TITLE
Can we get cookie values from the browser in route handlers?

### DISCUSSION THREAD

**Discussion by selcukgiray** [vercel/next.js#68173] [Help] [answered]
### Summary

```js
import { cookies } from "next/headers";

export async function GET() {
	const cookieStore = cookies();
	const { value: token } = cookieStore.get("token") ?? { value: null };

	const res = await fetch(process.env.API_URL + "me", {
		headers: {
			"Content-Type": "application/json",
			Authorization: `Bearer ${token}`,
		},
	});

	const data = await res.json();

	return Response.json(data);
}

```

I can create a new cookie in the route handlers, but I can't get the cookie value. What is the most logical way to get the cookie value in route handlers?

**Comment by Juneezee** [comment]
Your current code seems to be correct. What problem/error are you getting?

See **Route Handlers > Cookies** section [^1] for more info.

[^1]: https://nextjs.org/docs/app/building-your-application/routing/route-handlers#cookies

**Comment by icyJoseph** [comment]
For what domain and path are you setting the cookie? Could you should how the cookie looks like when set in the browser?

**Reply by selcukgiray** [reply]
**api/auth/login/route.js**
i can create the cookie succesfully after the login process.

```js
import { cookies } from "next/headers";
import { NextResponse } from "next/server";

export async function POST(request) {
	const cookieStore = cookies();

	const body = await request.json();

	const res = await fetch(process.env.API_URL + "login", {
		method: "POST",
		headers: {
			"Content-Type": "application/json",
		},
		body: JSON.stringify(body),
	});

	const { status, token, message } = await res.json();

	if (status && token) {
		cookieStore.set("token", token, {
			path: "/",
			expires: new Date(Date.now() + 3600 * 1000),
		});
	}

	return NextResponse.json({ status, message });
}

```

![cookie](https://github.com/user-attachments/assets/e4e5df60-61a9-40fd-a74f-abf20d971453)

![postman](https://github.com/user-attachments/assets/54365e09-1f53-45fc-9d3b-50e76385d94e)

```js
import { cookies } from "next/headers";

export async function GET() {
	const cookieStore = cookies();
	const { value: token } = cookieStore.get("token") ?? { value: null };
	console.log("tokennnn", token);
	const res = await fetch(process.env.API_URL + "me", {
		headers: {
			"Content-Type": "application/json",
			Authorization: `Bearer ${token}`,
		},
	});

	const data = await res.json();

	return Response.json(data);
}
```

I cant get the value of the cookie just getting a NULL value.

![image](https://github.com/user-attachments/assets/c5e13bd4-6d72-4626-bfea-6d725bf60f63)

I send a request with Postman and it gives 500 error.
 SyntaxError: Unexpected token '<', “<!DOCTYPE ”... is not valid JSON

![veriler](https://github.com/user-attachments/assets/3a68e400-e618-4ef7-a88f-e68b0c97676a)

When I go to api/auth/me/route.js I can see the data coming in.

**Reply by Juneezee** [reply, marked as answer]
@girayselcuk The problem you are facing seems to be similar to this: https://github.com/vercel/next.js/issues/52209#issuecomment-1741833812

**Reply by selcukgiray** [reply]
Thank you very much for your answers. The answer you sent was useful to me.
